src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,378 @@
+  { congr' 1, funext k,
+    rw [mul_comm _ (n : ℝ), mul_comm _ (n : ℝ), ←mul_assoc, ←mul_assoc],
+    congr' 1,
+    field_simp [h],
+    ring, },
+  { ring, },
+end
+
+end bernstein
+
+open bernstein
+
+local postfix `/ₙ`:2000 := z
+
+/--
+The `n`-th approximation of a continuous function on `[0,1]` by Bernstein polynomials,
+given by `∑ k, f (k/n) * bernstein n k x`.
+-/
+def bernstein_approximation (n : ℕ) (f : I →ᵇ ℝ) : I →ᵇ ℝ :=
+∑ k : fin (n+1), f k/ₙ • bernstein n k
+
+namespace bernstein_approximation
+
+@[simp] lemma apply (n : ℕ) (f : I →ᵇ ℝ) (x : I) :
+  bernstein_approximation n f x = ∑ k : fin (n+1), f k/ₙ * bernstein n k x :=
+by simp [bernstein_approximation]
+
+/--
+The set of points `k` so `k/n` is within `δ` of `x`.
+-/
+def S (f : I →ᵇ ℝ) (ε : ℝ) (h : 0 < ε) (n : ℕ) (x : I) : finset (fin (n+1)) :=
+{ k : fin (n+1) | dist k/ₙ x < f.modulus (ε/2) (half_pos h) }.to_finset
+
+/--
+If `k ∈ S`, then `f(k/n)` is close to `f x`.
+-/
+lemma lt_of_mem_S
+  {f : I →ᵇ ℝ} {ε : ℝ} {h : 0 < ε} {n : ℕ} {x : I} {k : fin (n+1)} (m : k ∈ S f ε h n x) :
+  abs (f k/ₙ - f x) < ε/2 :=
+begin
+  apply f.dist_lt_of_dist_lt_modulus (ε/2) (half_pos h),
+  simpa [S] using m,
+end
+
+/--
+If `k ∉ S`, then as `δ ≤ abs (x - k/n)`, we have the inequality `1 ≤ δ^-2 * (x - k/n)^2`.
+This particular formulation will be helpful later.
+-/
+lemma le_of_mem_S_compl
+  {f : I →ᵇ ℝ} {ε : ℝ} {h : 0 < ε} {n : ℕ} {x : I} {k : fin (n+1)} (m : k ∈ (S f ε h n x)ᶜ) :
+  (1 : ℝ) ≤ (f.modulus (ε/2) (half_pos h))^(-2 : ℤ) * (x - k/ₙ) ^ 2 :=
+begin
+  simp only [finset.mem_compl, not_lt, set.mem_to_finset, set.mem_set_of_eq, S] at m,
+  field_simp,
+  erw [le_div_iff (pow_pos f.modulus_pos 2), one_mul],
+  apply sqr_le_sqr,
+  rw abs_eq_self.mpr (le_of_lt f.modulus_pos),
+  rw [dist_comm] at m,
+  exact m,
+end
+
+end bernstein_approximation
+
+open bernstein_approximation
+open bounded_continuous_function
+open filter
+
+open_locale topological_space
+
+/--
+This is the proof given in [Richard Beals' *Analysis, an introduction*][beals-analysis], §7D,
+and reproduced on wikipedia.
+-/
+theorem bernstein_approximation_uniform (f : I →ᵇ ℝ) :
+  tendsto (λ n : ℕ, bernstein_approximation n f) at_top (𝓝 f) :=
+begin
+  simp only [metric.nhds_basis_ball.tendsto_right_iff, metric.mem_ball, dist_eq_norm],
+  intros ε h,
+  let δ := f.modulus (ε/2) (half_pos h),
+  have nhds_zero := tendsto_const_div_at_top_nhds_0_nat (2 * ∥f∥ * δ ^ (-2 : ℤ)),
+  filter_upwards [nhds_zero.eventually (gt_mem_nhds (half_pos h)), eventually_gt_at_top 0],
+  intros n nh npos',
+  have npos : 0 < (n:ℝ) := by exact_mod_cast npos',
+  -- Three easy inequalities we'll need later:
+  have w₀ : 0 ≤ ε / 2 := le_of_lt (half_pos h),
+  have w₁ : 0 ≤ 2 * ∥f∥ := mul_nonneg (by norm_num) (norm_nonneg f),
+  have w₂ : 0 ≤ 2 * ∥f∥ * δ^(-2 : ℤ) := mul_nonneg w₁ pow_minus_two_nonneg,
+  -- As `[0,1]` is compact, it suffices to check the inequality pointwise.
+  apply bounded_continuous_function.norm_lt_of_compact,
```

Comment:
You can actually do without this lemma (using `norm_le` instead) if you use the "for all `ε` there exists .... such that `|...| ≤ ε`" characterization of the limit, rather than the `|...| < ε` characterization.  You'd get this by putting `nhds_basis_closed_ball` in the first line rather than `nhds_basis_ball`.  Doesn't really matter though ....

Reply:
I think I'll leave this as is.